internal/provider/resource_vault_cluster_test.go

```diff
@@ -0,0 +1,136 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+)
+
+var (
+	testAccVaultClusterConfig = fmt.Sprintf(`
+resource "hcp_hvn" "test" {
+	hvn_id         = "test-hvn"
+	cloud_provider = "aws"
+	region         = "us-west-2"
+}
+	
+resource "hcp_vault_cluster" "test" {
+	cluster_id            = "test-vault-cluster"
+	hvn_id                = hcp_hvn.test.hvn_id
+	min_vault_version     = "v1.7.0"
+}
+`)
+)
+
+func TestAccVaultCluster(t *testing.T) {
+	resourceName := "hcp_vault_cluster.test"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testAccCheckVaultClusterDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testConfig(testAccVaultClusterConfig),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckVaultClusterExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "cluster_id", "test-vault-cluster"),
+					resource.TestCheckResourceAttr(resourceName, "hvn_id", "test-hvn"),
+					resource.TestCheckResourceAttr(resourceName, "tier", "DEV"),
+					resource.TestCheckResourceAttr(resourceName, "cloud_provider", "aws"),
+					resource.TestCheckResourceAttr(resourceName, "region", "us-west-2"),
+					resource.TestCheckResourceAttr(resourceName, "public_endpoint", "false"),
+					resource.TestCheckResourceAttr(resourceName, "namespace", "admin"),
+					resource.TestCheckResourceAttr(resourceName, "vault_version", "v1.7.0"),
```

Comment:
Depending on the above question about specifying `min_vault_version` in the test cluster config, the hardcoded version should maybe be removed. Looks like that would be consistent with the Consul cluster acceptance tests too.